common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -840,7 +846,7 @@ protected boolean confirmShutdown() {
         if (nanoTime - lastExecutionTime <= gracefulShutdownQuietPeriod) {
             // Check if any tasks were added to the queue every 100ms.
             // TODO: Change the behavior of takeTask() so that it returns on timeout.
-            wakeup(true);
+            taskQueue.offer(WAKEUP_TASK);
             try {
                 Thread.sleep(100);
```

Comment:
do we still need this sleep?

Reply:
@johnou I think so yes since it's how the graceful shutdown / quiet period is currently implemented. That said polling with `sleep` for this does seem unnecessarily cumbersome (as hinted by the TODO comment above)... probably something for a follow-on improvement.